addons/crm/models/crm_team.py

```diff
@@ -612,6 +619,8 @@ def _assign_and_convert_leads(self, force_quota=False):
                 member: {"assigned": self.env["crm.lead"], "quota": quota_per_member[member]}
                 for member in members_to_assign
             })
+            # Avoid use complete recordset given by read_group to avoid fetching all the lead in memory
+            leads_to_assign = self.env['crm.lead'].browse(leads_to_assign_ids).exists()
```

Comment:
exists because the ids have been read by _read_group and we may try to read them many transaction later. Some of them may have disappeared

Reply:
Could be worth to add a small comment in code maybe to avoid removing that code in some potential future refactoring ?